src/plan_ordered_append.c

```diff
@@ -0,0 +1,153 @@
+	 * doublecheck for those this way, but we are only dealing with single
+	 * relations at this level anyway so this should always match
+	 */
+	Assert(tle->resjunk == false || ht->space->main_table_relid != tle->resorigtbl);
+
+	/*
+	 * we only support direct column references for now
+	 */
+	if (!IsA(tle->expr, Var))
+		return false;
+
+	/* check dimension column is our ORDER BY column */
+	column = strVal(list_nth(rte->eref->colnames, AttrNumberGetAttrOffset(castNode(Var, tle->expr)->varattno)));
+	if (namestrcmp(&ht->space->dimensions[0].fd.column_name, column) != 0)
+		return false;
+
+	/*
+	 * check sort operation is either less than or greater than
+	 */
+	tce = lookup_type_cache(castNode(Var, tle->expr)->vartype, TYPECACHE_LT_OPR | TYPECACHE_GT_OPR);
+	if (sort->sortop != tce->lt_opr && sort->sortop != tce->gt_opr)
+		return false;
+
+	if (reverse != NULL)
+		*reverse = sort->sortop == tce->lt_opr ? false : true;
+
+	return true;
+}
+
+/*
+ * we use an existing MergeAppendPath here as starting point for creating
+ * our ordered AppendPath because it has all the required information we
+ * need to create our path. If pathkeys does not match the ORDER BY then
+ * we return the original MergeAppendPath
+ */
+Path *
+ts_ordered_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht, MergeAppendPath *merge)
+{
+	ListCell   *lc;
+	List	   *sorted = NIL;
+	AppendPath *append;
+	bool		parallel_safe = rel->consider_parallel;
+
+	/*
+	 * double check pathkeys of the MergeAppendPath actually is compatible
+	 * with the order supplied in the query since that is what our children
+	 * will be ordered by.
+	 */
+	if (!pathkeys_contained_in(root->sort_pathkeys, merge->path.pathkeys))
+		return (Path *) merge;
+
+	/* create subpaths for our append node */
+	foreach(lc, merge->subpaths)
+	{
+		Path	   *child = lfirst(lc);
+
+		/*
+		 * AppendPath is parallel_safe if all children are parallel safe
+		 */
+		parallel_safe = parallel_safe && child->parallel_safe;
+
+		/*
+		 * If the subpath already has an order that is equal or better than
+		 * the desired order for example from an index scan, then we can use
+		 * subpath directly otherwise we create a sort path on top of the
+		 * subpath
+		 */
+		if (pathkeys_contained_in(merge->path.pathkeys, child->pathkeys))
+		{
+			sorted = lappend(sorted, child);
+		}
+		else
+		{
+			sorted = lappend(sorted, create_sort_path(root, child->parent, child, merge->path.pathkeys, merge->limit_tuples));
+		}
+	}
+
+	/*
+	 * we set subpaths as NIL initially to skip PostgreSQL cost calculation
+	 * for children
+	 */
+#if PG96
+	append = create_append_path(rel, NIL, PATH_REQ_OUTER(&merge->path), 0);
+#elif PG10
+	append = create_append_path(rel, NIL, PATH_REQ_OUTER(&merge->path), 0, merge->partitioned_rels);
+#else
+	append = create_append_path(root, rel, NIL, NULL, PATH_REQ_OUTER(&merge->path), 0, false, merge->partitioned_rels, root->limit_tuples);
+#endif
```

Comment:
Why not create a `create_append_path_compat`?

Reply:
i didnt do it cause it would still need #ifdef on callsite because of partitioned_rels but if you prefer compat function i can do that

Reply:
ok leaving as is